fs_gcsfs/_gcsfs.py

```diff
@@ -70,6 +74,9 @@ def __init__(self,
         self.client = client
         if self.client is None:
             self.client = Client()
```

Comment:
I would still use `Client(_http=adapter)` and nor `Client._http = adapter`.

Reply:
IMHO it's the exact same kind of "invalid" API usage.

However, what we actually do is `self.client._http.mount("https://", adapter)`: We do not change the original adapter that the Client creates, but just mount an aditional adapter, so I'd say it's less intrusive ;)

Reply:
I disagree, `_http` could be intentionally be modified by the constructor before setting it to the private attribute self._http. So conditional on the necessity to manually specify the adapter the former way is cleaner. 
As for adding it as an additional mount, this exceeds my very limited knowledge of `request`.